File: 1079. Letter Tile Possibilities/main.go
package main

func numTilePossibilities(tiles string) int {
	count := make(map[rune]int)
	for _, c := range tiles {
		count[c]++
	}

	var backtrack func() int
	backtrack = func() int {
		res := 0
		for c := range count {
			if count[c] == 0 {
				continue
			}
			count[c]--
			res++
			res += backtrack()
			count[c]++
		}
		return res
	}

	return backtrack()
}

File: 1079. Letter Tile Possibilities/main.py
class Solution:
    def numTilePossibilities(self, tiles: str) -> int:
        used = set()

        def dfs(i, cur, visit):
            used.add(cur)
            visit.add(i)
            for j in range(len(tiles)):
                if j in visit: continue
                dfs(j, cur + tiles[j], visit)
            visit.remove(i)    
        
        for i in range(len(tiles)):
            dfs(i, tiles[i], set())

        return len(used)